opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -290,12 +301,20 @@ namespace Opm
                     perf_dis_gas_rate = getValue(dis_gas);
                     perf_vap_oil_rate = getValue(vap_oil);
                 }
-                if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx)) {
-                    const unsigned waterCompIdx = Indices::canonicalToActiveComponentIndex(FluidSystem::waterCompIdx);
-                    const Value vap_wat = rvw * cq_sGas;
-                    cq_s[waterCompIdx] += vap_wat;
-                    if (this->isProducer())
-                        perf_vap_wat_rate = getValue(vap_wat);
+            }
+
+            if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx) && FluidSystem::phaseIsActive(FluidSystem::gasPhaseIdx)) {
+                const unsigned waterCompIdx = Indices::canonicalToActiveComponentIndex(FluidSystem::waterCompIdx);
+                const unsigned gasCompIdx = Indices::canonicalToActiveComponentIndex(FluidSystem::gasCompIdx);
+                const Value cq_sWat = cq_s[waterCompIdx];
+                const Value cq_sGas = cq_s[gasCompIdx];
+                const Value vap_wat = rvw * cq_sGas;
+                const Value dis_gas_wat = rsw * cq_sWat;
+                cq_s[waterCompIdx] += vap_wat;
+                cq_s[gasCompIdx]   += dis_gas_wat;
+                if (this->isProducer()) {
+                    perf_vap_wat_rate = getValue(vap_wat);
```

Comment:
I think it is good that you combine these two features in this one block. To be clear: this is a bugfix for vaporized water, right? (Moving it out of the block requiring oil to be an active phase.)

Reply:
Yes I think so, but I have not checked. The perf_vap_wat_rate seems to be recomputed for producers at the end of this method. Maybe this whole block should be removed?

Reply:
Seems correct, assuming you mean lines 317-319.